lighthouse-core/test/gather/gather-runner-test.js

```diff
@@ -511,6 +511,30 @@ describe('GatherRunner', function() {
         });
   });
 
+  describe('#assertPageLoaded', () => {
+    it('passes when the page is loaded', () => {
+      const url = 'http://the-page.com';
+      const records = [{url}];
+      GatherRunner.assertPageLoaded(url, {online: true}, records);
+    });
+
+    it('throws when page fails to load', () => {
+      const url = 'http://the-page.com';
+      const records = [{url, failed: true, localizedFailDescription: 'foobar'}];
+      assert.throws(() => {
+        GatherRunner.assertPageLoaded(url, {online: true}, records);
+      }, /Unable.*foobar/);
+    });
+
+    it('throws when page times out', () => {
+      const url = 'http://the-page.com';
+      const records = [];
+      assert.throws(() => {
+        GatherRunner.assertPageLoaded(url, {online: true}, records);
+      }, /Unable.*timeout/);
+    });
+  });
```

Comment:
I've noticed we have some differing philosophies when it comes to `describe` blocks and test organization in general, haha, so I'm opposed but will defer to you if you want 'em moved :)